.github/workflows/python.yml

```diff
@@ -36,20 +39,36 @@ jobs:
         key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('Pipfile.lock') }}
       if: matrix.python-version == 'pypy3'
     - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install .[dev]
-    - name: Lint with flake8
-      run: |
-        # Use settings from mesas .flake8 file
-        flake8 . --count --show-source --statistics
-    - name: Lint with black
-      if: matrix.python-version == '3.8'
-      run: |
-        pip install black
-        black --check --exclude=mesa/cookiecutter-mesa/* .
+      run: pip install .[dev]
     - name: Test with pytest
-      run: |
-        pytest --cov=mesa tests/ --cov-report=xml
+      run: pytest --cov=mesa tests/ --cov-report=xml
     - name: Codecov
       uses: codecov/codecov-action@v1.0.15
+
+  lint-flake:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: pip install .[dev]
+    - name: Lint with flake8
+      # Use settings from mesas .flake8 file
+      run: flake8 . --count --show-source --statistics
+
+  lint-black:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: pip install .[dev]
```

Comment:
Is this line necessary for linting? The following line already does the Black installation.

Reply:
Good catch. Is there a reason why we wouldn't just add the black[juypter] to dev setup? And get rid of the second line?

Reply:
It may take longer time to install all the packages in `.[dev]`. It is better to just install the required dependencies to save time.

Reply:
Especially since the dependencies are not cached in this job.

Reply:
Fair point. 
@EwoutH can you adjust?

Reply:
Thanks, good catch!